### PR TITLE
Fix `ListViewAccessibleObject` control type to be dependent on ListView View type

### LINKED ISSUE #10546: The "Input Mask" dialog for MaskedTextBox have wrong Control in the DemoConsole application
### .NET version

9.0.0-alpha.1.23621.3

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, the MaskedTextBox start support from 9.0

### Issue description

The "Input Mask" dialog for MaskedTextBox have incorrect ControlType - LIst instead of Table

### Steps to reproduce

1. Test sample: [DemoConsole](https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/DemoConsole) in Winforms repo
2. Open the inspect tool to observe the ControlType of the "Input Mask" dialog for MaskedTextBox 

**Actually Results:**
The "Input Mask" dialog for MaskedTextBox have incorrect ControlType
image

**Expected Results:**
The "Input Mask" dialog for MaskedTextBox have correct ControlType
image

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #10546 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10560)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/MaskDesignerDialog.cs [hunk 1]
```diff
@@ -156,15 +156,16 @@ private void InitializeComponent()
...
+        [
         _maskDescriptionHeader,
         _dataFormatHeader,
         _validatingTypeHeader
-        });
+        ]);
         _listViewCannedMasks.FullRowSelect = true;
         _listViewCannedMasks.HideSelection = false;
         _listViewCannedMasks.Margin = new Padding(0, 3, 0, 3);
+        _listViewCannedMasks.AccessibleRole = AccessibleRole.Table;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/MaskDesignerDialog.cs:168]: 1. Once a control is declared as a Table, accessibility tools will be looking for patterns, events and properties that a table implements - https://learn.microsoft.com/dotnet/framework/ui-automation/ui-automation-support-for-the-table-control-type. We might have to add some patterns to ListView. You should be able to see missing patterns in Accessibility Insights tool when you point it to this ListView
2. It makes sense to make ControlType dependant on the ListVire.View ptoperty value, i.e. if View is Details, then ControlType is always Table

**Comment by Tanya-Solyanik** [comment]
ControlType change looks good. Do you need help with patterns or Accessibility insights?

**Comment by elachlan** [comment]
I ran Accessibility insights and the only issue it found was on the row for "<custom>" the Data Format was missing a "Name" because its an empty value. Our code is setting the name of the items as the text of the item. 

I wasn't sure on the missing patterns, I couldn't see it in the issues when I tested the control with Accessibility insights.

**Comment by SimonZhao888** [comment]
> I ran Accessibility insights and the only issue it found was on the row for "" the Data Format was missing a "Name" because its an empty value. Our code is setting the name of the items as the text of the item.

This is “by design”, Textbox will pick up Accessible name from the previous(in tab order) labels, or be set by the developer on their own initiative.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.ListViewAccessibleObject.cs [hunk 2]
```diff
@@ -244,7 +256,7 @@ internal override VARIANT GetPropertyValue(UIA_PROPERTY_ID propertyID)
                 // And we don't have a 100% guarantee it will be correct, hence set it ourselves.
                 UIA_PROPERTY_ID.UIA_ControlTypePropertyId when
                     this.GetOwnerAccessibleRole() == AccessibleRole.Default
-                    => (VARIANT)(int)UIA_CONTROLTYPE_ID.UIA_ListControlTypeId,
+                    => (VARIANT)(int)GetControlType(),
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.ListViewAccessibleObject.cs:259]:
```suggestion
                    => (VARIANT)(int)((this.TryGetOwnerAs(out ListView? owningListView) && owningListView.View == View.Details)
    ? UIA_CONTROLTYPE_ID.UIA_TableControlTypeId
    : UIA_CONTROLTYPE_ID.UIA_ListControlTypeId),
```

Comment (elachlan) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.ListViewAccessibleObject.cs]: Done.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Comment by elachlan** [comment]
Test failure seems unrelated:
```
System.Windows.Forms.Tests.ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(pixelFormat: Format32bppArgb, pixel00Color: Color [Red], givenPixel10Color: Color [A=200, R=50, G=75, B=100], expectedPixel10Color: Color [A=200, R=67, G=81, B=96]) [FAIL]
      Assert.Equal() Failure: Values differ
      Expected: Color [A=200, R=67, G=81, B=96]
      Actual:   Color [A=200, R=85, G=104, B=123]
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageList.ImageCollectionTests.cs(210,0): at System.Windows.Forms.Tests.ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(PixelFormat pixelFormat, Color pixel00Color, Color givenPixel10Color, Color expectedPixel10Color)
           at InvokeStub_ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(Object, Span`1)
           at System.Reflection.MethodBaseInvoker.InvokeWithFewArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    System.Windows.Forms.Tests.ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(pixelFormat: Format32bppPArgb, pixel00Color: Color [Red], givenPixel10Color: Color [A=200, R=50, G=75, B=100], expectedPixel10Color: Color [A=200, R=67, G=81, B=96]) [FAIL]
      Assert.Equal() Failure: Values differ
      Expected: Color [A=200, R=67, G=81, B=96]
      Actual:   Color [A=200, R=85, G=104, B=123]
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageList.ImageCollectionTests.cs(210,0): at System.Windows.Forms.Tests.ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(PixelFormat pixelFormat, Color pixel00Color, Color givenPixel10Color, Color expectedPixel10Color)
           at InvokeStub_ImageCollectionTests.ImageCollection_Item_Get32bppColorDepth_Success(Object, Span`1)
           at System.Reflection.MethodBaseInvoker.InvokeWithFewArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    System.Windows.Forms.Tests.CommonDialogTests.ShowDialog_ControlOwnerWithHandleWithVisualStyles_ReturnsExpected [SKIP]
      Crash with AbandonedMutexException. See: https://github.com/dotnet/arcade/issues/5325
    System.Windows.Forms.Tests.CommonDialogTests.ShowDialog_NonControlOwnerWithVisualStyles_ReturnsExpected [SKIP]
      Crash with AbandonedMutexException. See: https://github.com/dotnet/arcade/issues/5325
    System.Windows.Forms.Tests.CommonDialogTests.ShowDialog_ControlOwnerWithVisualStyles_ReturnsExpected [SKIP]
      Crash with AbandonedMutexException. See: https://github.com/dotnet/arcade/issues/5325
    System.Windows.Forms.Tests.ToolTipTests.ToolTip_SetToolTip_TabControl_DoesNotAddToolForTabControlItself [FAIL]
      Assert.Equal() Failure: Values differ
      Expected: 2
      Actual:   0
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolTipTests.cs(888,0): at System.Windows.Forms.Tests.ToolTipTests.ToolTip_SetToolTip_TabControl_DoesNotAddToolForTabControlItself()
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
           at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
    System.Windows.Forms.Tests.ToolTipTests.ToolTip_RemoveAll_Invokes_RemoveToolTip_OfControl [FAIL]
      System.InvalidOperationException : Adding the tip to the native ToolTip control did not succeed.
      Stack Trace:
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(839,0): at System.Windows.Forms.ToolTip.SetToolInfo(Control control, String caption)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(855,0): at System.Windows.Forms.ToolTip.CreateRegion(Control control)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(797,0): at System.Windows.Forms.ToolTip.CreateAllRegions()
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(528,0): at System.Windows.Forms.ToolTip.get_TopLevelControl()
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(846,0): at System.Windows.Forms.ToolTip.CreateRegion(Control control)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(638,0): at System.Windows.Forms.ToolTip.HandleCreated(Object sender, EventArgs eventargs)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(1236,0): at System.Windows.Forms.ToolTip.SetToolTipInternal(Control control, TipInfo info)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs(1208,0): at System.Windows.Forms.ToolTip.SetToolTip(Control control, String caption)
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolTipTests.cs(815,0): at System.Windows.Forms.Tests.ToolTipTests.ToolTip_RemoveAll_Invokes_RemoveToolTip_OfControl()
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
           at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
Unhandled exception. System.InvalidOperationException: Adding the tip to the native ToolTip control did not succeed.
   at System.Windows.Forms.ToolTip.SetToolInfo(Control control, String caption) in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 839
   at System.Windows.Forms.ToolTip.CreateRegion(Control control) in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 855
   at System.Windows.Forms.ToolTip.CreateAllRegions() in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 797
   at System.Windows.Forms.ToolTip.get_TopLevelControl() in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 528
   at System.Windows.Forms.ToolTip.CreateRegion(Control control) in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 846
   at System.Windows.Forms.ToolTip.HandleCreated(Object sender, EventArgs eventargs) in /_/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs:line 638
   at System.Windows.Forms.Control.OnHandleCreated(EventArgs e) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 7808
   at System.Windows.Forms.Control.WmCreate(Message& m) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 12014
   at System.Windows.Forms.Control.WndProc(Message& m) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 13022
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Scrolling/ScrollableControl.cs:line 1506
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam) in /_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs:line 372
--- End of stack trace from previous location ---
   at System.Windows.Forms.Application.ThreadContext.OnThreadException(Exception ex) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadContext.cs:line 843
   at System.Windows.Forms.Application.OnThreadException(Exception t) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Application.cs:line 1076
   at System.Windows.Forms.Control.WndProcException(Exception e) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 13298
   at System.Windows.Forms.Control.ControlNativeWindow.OnThreadException(Exception e) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlNativeWindow.cs:line 57
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam) in /_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs:line 386
```

**Review by SimonZhao888 — APPROVED** [review]

**Pull request merged**